### ISSUE TITLE
readyToLogin.py does not handle a delegated slice that is provisioned by a delegee with AM API V3

### ISSUE THREAD

**Issue by luisanevers** [GENI-NSF/geni-tools#206]
This ticket is being written to track the last issue found in gcf-2.1 release testing.  This issue is minor and not likely to happen and specific to AM API V3.

In a scenario where the following takes place:

   - lnevers creates the slice and delegates it to lnevers2.
   - lnevers2 uses delegation file to allocate, provision and start the sliver.

In this case, the readyToLogin does not work for either of the 2 users, it reports "ERROR: No manifest found from describe", but it show the manifest as part of the output for both users.

Will provide omni commands, omni_config and readyToLogin.py output as soon as my test VMs are available.

 _Imported from trac ticket [#206](http://trac.gpolab.bbn.com/gcf/ticket/206),  created by lnevers on 10-09-2012 at 09:30, last modified: 12-17-2012 at 11:55_

**Comment by luisanevers** [comment]
_lnevers (luisanevers) attached [lnevers-omni.txt](http://trac.gpolab.bbn.com/gcf/attachment/ticket/206/lnevers-omni.txt) on 10-11-2012 at 08:02_

**Comment by luisanevers** [comment]
_lnevers (luisanevers) attached [omni_config-lnevers2](http://trac.gpolab.bbn.com/gcf/attachment/ticket/206/omni_config-lnevers2) on 10-11-2012 at 08:02_

**Comment by luisanevers** [comment]
_lnevers (luisanevers) attached [omni_config-lnevers](http://trac.gpolab.bbn.com/gcf/attachment/ticket/206/omni_config-lnevers) on 10-11-2012 at 08:02_

**Comment by luisanevers** [comment]
_lnevers (luisanevers) attached [lnevers2-readyToLogin.txt](http://trac.gpolab.bbn.com/gcf/attachment/ticket/206/lnevers2-readyToLogin.txt) on 10-11-2012 at 08:02_

**Comment by luisanevers** [comment]
_lnevers (luisanevers) attached [lnevers2-omni.txt](http://trac.gpolab.bbn.com/gcf/attachment/ticket/206/lnevers2-omni.txt) on 10-11-2012 at 08:02_

**Comment by luisanevers** [comment]
_lnevers (luisanevers) attached [lnevers-readyToLogin.txt](http://trac.gpolab.bbn.com/gcf/attachment/ticket/206/lnevers-readyToLogin.txt) on 10-11-2012 at 08:02_

**Comment by ahelsing** [comment]
If Omni redirected you to a different AM URL, then readyToLogin wouldn't find the entry - it looks up the result by the AM URL given originally. We might consider having readyToLogin accept the output even if the AM URL is different. E.G. 

```
if not apicallout.has_key(key)
  if len(apicallout.keys() == 1:
    print "WARN: Got result for AM URL %s instead of %s - did Omni redirect you?" % ....
  else:
    print "ERROR: ...." 
    sys.exit(-1)
```


_Trac comment by ahelsing on 10-09-2012 at 09:38_

**Comment by luisanevers** [comment]
Following is a log of the operations that lead up to this problem:

As user lnevers, create slice, get slice credentials, and delegate slice. Captured in attachment lnevers-omni.txt.

As user lnevers2, use delegation file to allocate, provision, and start the sliver. Captured in attachment lnevers2-omni.txt


Ran the readyToLogin.py script for both lnevers and lnevers2 accounts. See attached files lnevers-readyToLogin.txt and lnevers2-readyToLogin.txt.

Also attached the file omni_config-lnevers and omni_config-lnevers2.

___Question''__

The error reported by readyToLogin.py states: 

  ''ERROR: No manifest found from describe at https://www.emulab.net/protogeni/xmlrpc/am/2.0; review the logs._


What logs??


_Trac comment by lnevers (github user: luisanevers) on 10-11-2012 at 08:01_

**Comment by innnkkki** [comment]
Fixed at  [9b6b8580ed842b7f273e773f865ae493f1cd9f81](https://github.com/GENI-NSF/geni-tools/commit/9b6b8580ed842b7f273e773f865ae493f1cd9f81) but I haven't tested the whole delegation process. 

If problems persists please re-open. 

_Trac comment by nriga (github user: innnkkki) on 12-17-2012 at 11:55_
